retrieval/scrape.go

```diff
@@ -446,6 +461,28 @@ func (sl *scrapeLoop) stop() {
 	<-sl.done
 }
 
+func (sl *scrapeLoop) processScrapeResult(samples model.Samples, scrapeErr error, start time.Time) error {
+	// Collect samples post-relabelling and label handling in a buffer.
+	buf := &bufferAppender{buffer: make(model.Samples, 0, len(samples))}
```

Comment:
> you will not see all the accumulated buffer allocations of a full scrape cycle but only those that happened since the last GC run.

I was looking at total allocated.

> It would be interesting to see how much CPU is used in GC before and after.

No noticeable difference.